src/main/java/io/singularitynet/MissionHandlerInterfaces/IVideoProducer.java

```diff
@@ -46,7 +46,8 @@ enum VideoType
      * @return an array of bytes representing this frame.<br>
      * (The format is unspecified; it is up to the IVideoProducer implementation and the agent to agree on how the data is formatted.)
      */
-    Map.Entry<ByteBuffer, int[]> getFrame(MissionInit missionInit);
+
+    int[] getFrame(MissionInit missionInit, ByteBuffer buffer);
```

Comment:
```suggestion
    int[] writeFrame(MissionInit missionInit, ByteBuffer buffer);
```

it looks more like writeFrame now

Reply:
Renamed.